apps/prairielearn/src/components/QuestionsTable.html.ts

```diff
@@ -0,0 +1,207 @@
+import { html, HtmlSafeString } from '@prairielearn/html';
+import { nodeModulesAssetPath, compiledScriptTag } from '../lib/assets';
+import { EncodedData } from '@prairielearn/browser-utils';
+
+export function QuestionsTable(
+  questions,
+  showAddQuestionButton,
+  current_course_instance,
+  course_instances,
+  urlPrefix,
+  plainUrlPrefix,
+  errorMessage,
+): HtmlSafeString {
+  const has_legacy_questions = questions.some((row) => row.display_type !== 'v3');
+  const course_instance_ids = (course_instances || []).map((course_instance) => course_instance.id);
+  return html`
+    ${EncodedData(course_instance_ids, 'course-instance-ids')};
+    ${EncodedData(showAddQuestionButton, 'show-add-question-button')}
+    ${EncodedData(urlPrefix, 'url-prefix')} ${EncodedData(plainUrlPrefix, 'plain-url-prefix')}
+    <style>
+      .sticky-column {
+        position: sticky;
+        left: 0;
+        background: white;
+        background-clip: padding-box;
+        box-shadow: inset -1px 0 #dee2e6;
+      }
+      .table-hover tbody tr:hover td.sticky-column {
+        color: #212529;
+        background-color: #efefef;
+      }
+      .fixed-table-toolbar {
+        padding: 0 1em 0 1em;
+      }
+      .fixed-table-toolbar div.pagination,
+      .fixed-table-toolbar div.pagination-detail {
+        margin: 0 1em 0 0 !important;
+      }
+    </style>
+
+    <script src="${nodeModulesAssetPath('lodash/lodash.min.js')}"></script>
+    <script src="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.js')}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js',
+      )}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.js',
+      )}"></script>
+    <link
+      href="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.css')}"
+      rel="stylesheet"
+    />
+    <link
+      href="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.css',
+      )}"
+      rel="stylesheet"
+    />
+    <link
+      href="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.css',
+      )}"
+      rel="stylesheet"
+    />
+
+    ${compiledScriptTag('questionsTable.js')}
```

Comment:
Let's avoid inlining these into the component itself; it should be up to the page using this component to load the `questionsTable.js` script. As discussed elsewhere, we should also try to import any needed JS directly into `questionsTable.js`. As for the styles: we also have the ability to bundle styles. See `assets/stylesheets`; you can use `@import` to import things into the file.

Reply:
I've fixed the lodash problems. 

For Bootstrap-table, it really doesn't seem like they have a friendly way to just `import` the code in javascript, the docs make no mention of it: https://bootstrap-table.com/docs/getting-started/introduction/. If I must import it using tags like this, where would you like me to put them? 

Re: Stylesheets, it is not clear to me what you are asking me to do. Do you want me to have a stylesheet in `assets/stylesheets` that imports all the styles needed for this page, and then have that one sheet be imported somewhere else? if so where?

Reply:
If I do:
```js
import 'bootstrap-table/dist/bootstrap-table.min.js';
import 'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js';
import 'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.js';
```
at the top of  `questionsTable.js`, I get 

```
questionsTable.js:184 Uncaught TypeError: $(...).bootstrapTable is not a function
    at questionsTable.js:184:24
```

I haven't gotten to trying the CSS yet.

Reply:
Turns out Bootstrap Table is doing some extremely janky stuff that means that trying to import its JavaScript does bad things that don't work with how we're doing things:

https://github.com/wenzhixin/bootstrap-table/blob/5d3001609b63b3e16002850fef537717a4280eaa/dist/bootstrap-table.js#L1-L5

Given that, you can keep loading these things independently!

Reply:
Wow, that looks extremely janky indeed. Thanks for looking into that for me. I'll go ahead and leave a comment so its clear that its not the preferred way to do things.

Reply:
good news is, importing the CSS the proper way works fine